bentoml/_internal/utils/analytics/usage_stats.py

```diff
@@ -0,0 +1,170 @@
+import os
+import typing as t
+import logging
+import secrets
+import platform
+import threading
+from typing import TYPE_CHECKING
+from datetime import datetime
+from datetime import timezone
+from functools import wraps
+from functools import lru_cache
+
+import yaml
+import psutil
+import requests
+from simple_di import inject
+from simple_di import Provide
+
+from bentoml import __version__ as BENTOML_VERSION
+
+from ...configuration.containers import CLIENT_ID_PATH
+from ...configuration.containers import BentoMLContainer
+
+if TYPE_CHECKING:
+    P = t.ParamSpec("P")
+    T = t.TypeVar("T")
+    AsyncFunc = t.Callable[P, t.Coroutine[t.Any, t.Any, t.Any]]
+
+logger = logging.getLogger(__name__)
+
+BENTOML_DO_NOT_TRACK = "BENTOML_DO_NOT_TRACK"
+BENTOML_TRACKING_URL = "https://t.bentoml.com"
+BENTOML_USAGE_REPORT_INTERVAL_SECONDS = "__BENTOML_USAGE_REPORT_INTERVAL_SECONDS"
```

Comment:
instead of using env var, you can always monkey patching a constant in a module

Reply:
noted